src/emitters/directional.cpp

```diff
@@ -89,13 +91,33 @@ class DirectionalEmitter final : public Emitter<Float, Spectrum> {
                       unpolarized<Spectrum>(0.f));
     }
 
-    std::pair<Ray3f, Spectrum> sample_ray(Float /*time*/,
-                                          Float /*wavelength_sample*/,
-                                          const Point2f & /*sample2*/,
+    std::pair<Ray3f, Spectrum> sample_ray(Float time,
+                                          Float wavelength_sample,
+                                          const Point2f & sample2,
                                           const Point2f & /*sample3*/,
                                           Mask active) const override {
         MTS_MASKED_FUNCTION(ProfilerPhase::EndpointSampleRay, active);
+
         NotImplementedError("sample_ray");
+        // TODO: finish this
+
+        // 1. Sample spectrum
+        auto [wavelengths, weight] =
+            sample_wavelength<Float, Spectrum>(wavelength_sample);
+
+        // 2. Sample spatial component
+        // TODO: sample a point on such that cast ray will hit the scene
+        Vector3f v0 = warp::square_to_uniform_sphere(sample2);
```

Comment:
This could should be similar to the Mitsuba 1 implementation (sample on disk and transform)

Reply:
See latest commits for an implementation.